h2o-py/h2o/group_by.py

```diff
@@ -17,25 +17,47 @@ class GroupBy(object):
     """
     A class that represents the group by operation on an H2OFrame.
 
+    The returned groups are sorted by the natural group-by column sort.
+
+    :param H2OFrame fr: H2OFrame that you want the group by operation to be performed on.
+    :param by: by can be a str, a list of str, an int or a list of int which denote the column name(s) or
+        column index(indices) of column(s) to group on.
+
     Sample usage:
 
        >>> my_frame = ...  # some existing H2OFrame
        >>> grouped = my_frame.group_by(by=["C1", "C2"])
        >>> grouped.sum(col="X1", na="all").mean(col="X5", na="all").max()
        >>> grouped.get_frame()
 
-    Any number of aggregations may be chained together in this manner.
+    Any number of aggregations may be chained together in this manner.  Note that once the aggregation
+    operations are complete, calling the GroupBy object with a new set of aggregations will yield no
+    effect.  You must generate a new GrouBy object in order to apply a new aggregation on it.  In addition,
+    certain aggregations are only defined for numerical or categorical columns.  An error will be thrown for
+    calling aggregation on the wrong data types.
 
     If no arguments are given to the aggregation (e.g. "max" in the above example),
     then it is assumed that the aggregation should apply to all columns but the
     group by columns.
 
-    The ``na`` parameter is one of ``"all"`` (include NAs), ``"ignore"``, ``"rm"`` (exclude NAs).
+    The ``na`` parameter is one of ``"all"`` (include NAs), ``"ignore"``, ``"rm"`` (exclude NAs).  It provides
```

Comment:
All GroupBy aggregations take parameter ``na``, which controls treatment of NA values during the calculation. It can be one of:

  * ``all`` (default) -- any NAs are used in the calculation as-is; which usually results in the final result being NA too
  * ``ignore`` -- NA entries are not included in calculations, but the total number of entries is taken as the total number of rows. For example, ``mean([1, 2, 3, nan], na="ignore")`` will produce `1.5`.
  * ``rm`` -- NA entries are skipped during the calculations, reducing the total effective count of entries. For example, ``mean([1, 2, 3, nan], na="rm")`` will produce `2`.